src/app/controllers/user_controller.py

```diff
@@ -1,18 +1,50 @@
 from django.contrib.auth import get_user_model
 from rest_framework import viewsets
-
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework.decorators import action
+from rest_framework.permissions import AllowAny
+from rest_framework import status
 from app.models.user import User
 from app.serializers.user_serializer import UserSerializer
 
 User = get_user_model()
 
+class UserViewSet(viewsets.ModelViewSet):
+    queryset = User.objects.all()
+    serializer_class = UserSerializer
+
+    def get_permissions(self):
+        if self.request.method == 'POST':
+            self.permission_classes = [AllowAny,]
+        return super(UserViewSet, self).get_permissions()
+
+    def perform_create(self, serializer):
+        if serializer.is_valid():
+            user = serializer.save()
+            if user:
+                return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
If I'm not mistaken, this method is included in the serializer by default. Since we're not adding any additional details during user creation, we can actually straight up remove this.